src/JsonWebTokenDynamicValue.js

```diff
@@ -58,9 +58,18 @@ class JsonWebTokenDynamicValue {
       payload = { ...this.payload }
     }
 
-    const secret = this.signatureSecretIsBase64
-      ? {b64: jsrsasign.b64utob64(this.signatureSecret)}
-      : {utf8: this.signatureSecret}
+    let secret;
+
+    if(this.alg.substring(0, 2) == "HS") {
+      secret = this.signatureSecretIsBase64
+        ? {b64: jsrsasign.b64utob64(this.signatureSecret)}
+        : {utf8: this.signatureSecret}
+    } else {
+      secret = this.signatureSecretIsBase64
+        ? jsrsasign.b64utoutf8(this.signatureSecret)
+        : this.signatureSecret
+    }
```

Comment:
Great you've fixed this, tbh I couldn't find the right format for HS and RS algs in the jsrsasign doc. Thanks for looking this up.